pennylane/transforms/hamiltonian_expand.py

```diff
@@ -86,13 +81,36 @@ def hamiltonian_expand(tape, group=True):
             qml.PauliX(wires=2)
             qml.expval(H)
 
-        # split H into observable groups [qml.PauliZ(0)] and [qml.PauliX(1), qml.PauliX(0)]
-        tapes, fn = qml.transforms.hamiltonian_expand(tape)
-        print(len(tapes)) # 2
+    With grouping, the Hamiltonian gets split into two observable groups (here ``[qml.PauliZ(0)]`` and
+    ``[qml.PauliX(1), qml.PauliX(0)]``):
+    >>> tapes, fn = qml.transforms.hamiltonian_expand(tape)
+    >>> len(tapes)
+    2
+
+    Without grouping it gets split into three groups (``[qml.PauliZ(0)]``, ``[qml.PauliX(1)]`` and ``[qml.PauliX(0)]``):
+    >>> tapes, fn = qml.transforms.hamiltonian_expand(tape, group=False)
+    >>> len(tapes)
+    3
+
+    Alternatively, if the Hamiltonian has already computed groups, they are used even if ``group=False``:
+
+    .. code-block:: python3
+
```

Comment:
Very minor, line 99 here is long and causes a horizontal scrollbar in the rendered example. Could avoid this by defining coeffs/obs as variables first and then using them in construction of `H`.